### ISSUE TITLE
Bearer token not sent with request

### ISSUE THREAD

**Issue by eaidland** [paulvanbladel/aurelia-auth#19]
So I've updated Aurelia and the auth. Before I updated I could do this in the activate:
   return this.http.get(this.authService.config.baseUrl + this.api)
    .then(response => {
      console.log(response)
      this.employees = response.content;
    });

And the bearer token was sent With the request. Now it's empty. My authConfig haven't been changed. I can also see that my token is in the local Storage.

What have I missed? Something New in the Aurelia update?

Thanks in advance

![image](https://cloud.githubusercontent.com/assets/10772259/9737409/d2902c88-5646-11e5-97bb-b8f8618bfd25.png)

![image](https://cloud.githubusercontent.com/assets/10772259/9737449/05e04028-5647-11e5-95b1-7bad9e9b574f.png)

**Comment by paulvanbladel** [comment]
Yeah, I confirm here is something wrong.
Happens also in aurelia-auth-sample.
No clue what is happening here, The transformer that I add isn't called any longer :(
Maybe something changed in aurelia-http-client.
This needs closer inspection...

**Comment by eaidland** [comment]
Great!... or not great, but great that it just wasn't my app :+1: 

Thanks for quick response

**Comment by paulvanbladel** [comment]
If you have time (i haven't currently), could you please try to add a transformer to the http config and check it is still called ?

I'm doing following:
```
RequestBuilder.addHelper('authTokenHandling', ()=>{
			return (client, processor, message)=>{
				if (this.auth.isAuthenticated() && this.config.httpInterceptor) {
					var tokenName = this.config.tokenPrefix 
					? this.config.tokenPrefix + '_' + this.config.tokenName : this.config.tokenName;
					var token = this.storage.get(tokenName);

					if (this.config.authHeader && this.config.authToken) {
						token = this.config.authToken + ' ' + token;
					}

					message.headers.add(this.config.authHeader, token);
				}
			}
		});

		this.http.configure(x => {
		x.authTokenHandling();
      	x.withHeader('Accept', 'application/json');
```
of course the test transformer can also do a simple console.log (instead of adding a jwt token in header)

**Comment by eaidland** [comment]
_createClass(_default, [{
					key: 'configure',
					value: function configure() {
						var _this = this;
						 	console.log("TEST1");
						RequestBuilder.addHelper('authTokenHandling', function () {
							console.log("TEST2");
							return function (client, processor, message) {
								console.log("TEST2 " +  client);
								console.log("TEST2 " +processor );
								console.log("TEST2 " + message);
								console.log("TEST3");
								console.log("TEST3 " +_this.auth.isAuthenticated());
								console.log("TEST3 " +_this.config.httpInterceptor);
								if (_this.auth.isAuthenticated() && _this.config.httpInterceptor) {
									var tokenName = _this.config.tokenPrefix ? _this.config.tokenPrefix + '_' + _this.config.tokenName : _this.config.tokenName;
									var token = _this.storage.get(tokenName);
										console.log("token " + token);

									if (_this.config.authHeader && _this.config.authToken) {
										token = _this.config.authToken + ' ' + token;
											console.log("authHeader " + token);
									}

									message.headers.add(_this.config.authHeader, token);
								}
							};
						});

						this.http.configure(function (x) {
							x.authTokenHandling();
							x.withHeader('Accept', 'application/json');
						});
					}
				}]);

![image](https://cloud.githubusercontent.com/assets/10772259/9742056/3ed5b248-565e-11e5-95f5-af906cc97184.png)

**Comment by lukeskinner** [comment]
I found a fix for my local setup - worth checking to see if it fixes yours as well.

In my config.js file I had the following:
```
    "github:paulvanbladel/aurelia-auth@0.9.11": {
      "aurelia-framework": "github:aurelia/framework@0.15.0",
      "aurelia-http-client": "github:aurelia/http-client@0.10.3",
      "aurelia-router": "github:aurelia/router@0.11.0"
    },
```

After changing the http-client version number to 0.11.0 it seems to work again:
```
    "github:paulvanbladel/aurelia-auth@0.9.11": {
      "aurelia-framework": "github:aurelia/framework@0.15.0",
      "aurelia-http-client": "github:aurelia/http-client@0.11.0",
      "aurelia-router": "github:aurelia/router@0.11.0"
    },
```

I haven't had much experience in this area but presumably this is because aurelia-auth is still pointing to old versions of aurelia (see https://github.com/paulvanbladel/aurelia-auth/blob/master/config.js). So when you try to load it, it attempts to use the old 0.10.3 version of the http-client. I could be completely wrong though!

**Comment by eaidland** [comment]
Worked for me :+1:

**Comment by paulvanbladel** [comment]
Thanks @lukeskinner .
I probably need to update the framework references in aurelia-auth.
@eaidland  I'll check this and keep you posted.

**Comment by paulvanbladel** [comment]
Ok. solved with aurelia-auth 0.9.14.
fix via:
```
jspm uninstall aurelia-auth
jspm install aurelia-auth
```

**Comment by spideynr** [comment]
Just for a reference for future issues. I had the same issue with aurelia-auth 0.9.19
where i happened to have http-client 0.11.0 and newest aurelia-auth used http-client 0.12.0 

When i switched http-client 0.11.0 for 0.12.0 everything worked as expected. Thanks !
